modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -545,6 +546,20 @@ int iocshSetError(int err)
     return err;
 }
 
+void addPwdMacro(iocshContext *context) {
+    if (context != NULL) {
+        char buf[1024];
+        char *pwd = getcwd(buf, sizeof(buf) - 1);
+        if (pwd) {
+            macPutValue(context->handle, "PWD", pwd);
+        } else {
+            if (errno == ERANGE) {
+                perror("Current path is too long");
```

Comment:
With `perror()` there is no need to test `errno` explicitly.  A simple `perror("getcwd()");` would do.

Reply:
I tested it here but didn't like the result. Take a look:
- When the path is smaller than 1024, perror will print "<path name>: Success" every time addPwdMacro() is called. On my test IOC that happens 3 times.
- For big paths, it will print "Numerical result out of range" in the same 3 moments.

Don't you think a more descriptive text is better?

Reply:
>     * On my test IOC that happens 3 times.

Maybe an example will help.  I'm thinking of the follow, where `getcwd()` is only called when `getcwd()` indicates an error.

```c++;
        if (char *pwd = getcwd(buf, sizeof(buf) - 1)) {
            macPutValue(context->handle, "PWD", pwd);
        } else {
            perror("getcwd()");
        }
```

> Don't you think a more descriptive text is better?

```c
errno = ERANGE;
perror("Current path is too long");
```

For me prints `Current path is too long: Numerical result out of range`, which seems redundant.  `man 3 getcwd` also lists eg. `EINVAL` as a possible error condition.  This prints `Current path is too long: Invalid argument`, which seems contradictory.

The argument of `perror()` is meant to describe the operation which has failed.  `perror()` then adds information about why this operation has failed.

Reply:
Yes, looks like perror is not adequate at all to show these messages.